gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -98,24 +98,23 @@ public void setText (BitmapFont font, CharSequence str, int start, int end, Colo
 		glyphRunPool.freeAll(runs);
 		runs.clear();
 
+		for (int i = 1, n = colorStack.size; i < n; i++)
+			colorPool.free(colorStack.get(i));
+		colorStack.clear();
+
```

Comment:
In case of an empty String the pools would never be cleared (line 109), so I moved it up here.
Also I think the pools need to be freed and cleared even if `markupEnabled = false` as this could have been changed in the meantime or another BitmapFont could have been used.

Reply:
Might make sense to clear it at the end of a run, rather than at the start of the next one.